src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -228,6 +228,81 @@ class E
             comp.VerifyDiagnostics();
         }
 
+        
+        [Fact, WorkItem(49263, "https://github.com/dotnet/roslyn/issues/49263")]
+        public void DeriveFromSelf()
+        {
+            var src = @"
+record R : R;
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (2,8): error CS0146: Circular base type dependency involving 'R' and 'R'
+                // record R : R;
+                Diagnostic(ErrorCode.ERR_CircularBase, "R").WithArguments("R", "R").WithLocation(2, 8),
+                // (2,8): error CS0115: 'R.ToString()': no suitable method found to override
+                // record R : R;
+                Diagnostic(ErrorCode.ERR_OverrideNotExpected, "R").WithArguments("R.ToString()").WithLocation(2, 8),
+                // (2,8): error CS0115: 'R.EqualityContract': no suitable method found to override
+                // record R : R;
+                Diagnostic(ErrorCode.ERR_OverrideNotExpected, "R").WithArguments("R.EqualityContract").WithLocation(2, 8),
+                // (2,8): error CS0115: 'R.Equals(object?)': no suitable method found to override
+                // record R : R;
+                Diagnostic(ErrorCode.ERR_OverrideNotExpected, "R").WithArguments("R.Equals(object?)").WithLocation(2, 8),
+                // (2,8): error CS0115: 'R.GetHashCode()': no suitable method found to override
+                // record R : R;
+                Diagnostic(ErrorCode.ERR_OverrideNotExpected, "R").WithArguments("R.GetHashCode()").WithLocation(2, 8)
+                );
+        }
+
+        [CombinatorialData]
+        [Theory, WorkItem(49302, "https://github.com/dotnet/roslyn/issues/49302")]
+        public void GetSimpleNonTypeMembers(bool useCompilationReference)
+        {
+            var lib_src = @"
+public record RecordA(RecordB B);
+
+public record RecordB(int C);
+";
+            var lib_comp = CreateCompilation(lib_src);
+
+            var src = @"
+class C
+{
+    void M(RecordA a, RecordB b)
+    {
+        _ = a.B == b;
+    }
+}
+";
+            var comp = CreateCompilation(src, references: new[] { useCompilationReference ? lib_comp.ToMetadataReference() : lib_comp.EmitToImageReference() });
+            comp.VerifyEmitDiagnostics();
```

Comment:
Does this call `GetSimpleNonTypeMembers()`? Consider calling that method directly.

Same comment for test below.

Reply:
Both tests repro the issue if the fix isn't present. The one with multiple compilations is from the reported issue, the one with single compilation is more direct.

Reply:
> Does this call GetSimpleNonTypeMembers()? Consider calling that method directly.

Agree with this suggestion.

---
In reply to: [567031521](https://github.com/dotnet/roslyn/pull/50900#discussion_r567031521) [](ancestors = 567031521)